networkx/algorithms/lowest_common_ancestors.py

```diff
@@ -35,6 +35,22 @@ def tree_all_pairs_lowest_common_ancestor(G, root=None, pairs=None):
     lcas : generator of tuples `((u, v), lca)` where `u` and `v` are nodes
         in `pairs` and `lca` is their lowest common ancestor.
 
+    Examples
+    --------
+
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(1, 3), (2, 4), (1, 2)])
+    >>> dict(nx.tree_all_pairs_lowest_common_ancestor(G))
+    {(3, 3): 3, (3, 4): 1, (4, 4): 4, (3, 2): 1, (2, 2): 2, (4, 2): 2, (1, 1): 1, (3, 1): 1, (2, 1): 1, (4, 1): 1}
+
+     We can also use `pairs` argument to specify the pairs of nodes for which we
+     want to compute lowest common ancestors. Here is an example:
```

Comment:
It looks like the indentation level here differs from the examples. From a parsing perspective this probably doesn't matter, but from a readability perspective I think it's better if it matches with the examples!